lib/infrastructure/seed_generator_repository.dart

```diff
@@ -0,0 +1,70 @@
+import 'dart:convert';
+import 'package:dartz/dartz.dart';
+import 'package:dio/dio.dart';
+import 'package:injectable/injectable.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+import 'package:qr_generator/domain/failures/common_failure.dart';
+import 'package:qr_generator/infrastructure/dtos/seed_dto.dart';
+import 'package:qr_generator/domain/i_seed_generator_repository.dart';
+import 'package:qr_generator/domain/seed.dart';
+import 'package:qr_generator/infrastructure/extensions/dio_extensions.dart';
+
+@LazySingleton(as: ISeedGeneratorRepository)
+class SeedGeneratorRepository implements ISeedGeneratorRepository {
+  final SharedPreferences _sharedPreferences;
+  final Dio _dio;
+
+  static final endpoint = Uri.parse('http://mtask.mx/seed');
+  static const cacheKey = "seed";
+
+  SeedGeneratorRepository(
```

Comment:
**suggestion**: This could be const.

Reply:
You actually got me here, how can this be done? As my understanding Uri.parse does not have const constructors.
```
Const variables must be initialized with a constant value.
Try changing the initializer to be a constant expression.
```

and as for cacheKey I'm certain I cannot have them inside classes or else I get the following error: 
```
Only static fields can be declared as const.
Try declaring the field as final, or adding the keyword 'static'.
```
Do you mean declaring cacheKey outside the class? Like a file named 'constants.dart' with all my constants and importing it

Reply:
Sorry I was referring to the constructor being const.